STYLE.md

```diff
@@ -0,0 +1,125 @@
+# Ops Python style guide
+
+This is the Python style guide we use for the Ops project. It's also the style we're converging on for other projects maintained by the Charm Tech team.
+
+We use Ruff for formatting, and run our code through the Pyright type checker. We also try to follow [PEP 8](https://peps.python.org/pep-0008/), the official Python style guide. However, PEP 8 is fairly low-level, so in addition we've come up with the following style guidelines.
+
+New code should follow these guidelines, unless there's a good reason not to. Sometimes existing code doesn't follow these, but we're happy for it to be updated to do so (either all at once, or as you change nearby code).
+
+Of course, this is just a start! We add to this list as things come up in code review; this list reflects our team decision.
```

Comment:
High level comment: this reads like prose.

I wish it was:
- either prescriptive (do this)
- or as a statement (we have settled on this this / we currently do this)

Reply:
I tried fiddling with this text, but I somewhat disagree -- these are introductory paragraphs providing context on why we have this document. "We use Ruff" is descriptive, but Ruff (and the like) only gets you so far, hence the need for this style guide.